runtime/js/40_command.js

```diff
@@ -0,0 +1,168 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+"use strict";
+
+((window) => {
+  const core = window.Deno.core;
+  const { assert, pathFromURL } = window.__bootstrap.util;
+  const { read } = window.__bootstrap.io;
+  const { writeAll } = window.__bootstrap.buffer;
+  const { illegalConstructorKey } = window.__bootstrap.webUtil;
+  const { ArrayPrototypeMap, ObjectEntries, String } =
+    window.__bootstrap.primordials;
+
+  class Command {
+    #args;
+
+    constructor(command, {
+      args = [],
+      cwd = undefined,
+      clearEnv = false,
+      env = {},
+    } = {}) {
+      this.#args = {
+        cmd: pathFromURL(command),
+        args: ArrayPrototypeMap(args, String),
+        cwd: pathFromURL(cwd),
+        clearEnv,
+        env: ObjectEntries(env),
+      };
+    }
+
+    spawn(options = {}) {
+      const child = core.opSync("op_command_spawn", {
+        ...this.#args,
+        ...options,
+      });
+
+      return new Child(illegalConstructorKey, child);
+    }
+
+    async status(options = {}) {
+      return await core.opAsync("op_command_status", {
+        ...this.#args,
+        ...options,
+      });
+    }
+
+    async output() {
+      return await core.opAsync("op_command_output", this.#args);
+    }
+  }
+
+  class Child {
```

Comment:
It would be great if `Child` could do cleanup of "related" resource, ie. close `stdin`, `stdout` and `stderr` once they are no longer needed/were consumed by Rust APIs